File: tests/integration/transforms/python/simple/random_.py
# Copyright 2016-2021, KhulnaSoft Ltd.  All rights reserved.

from typing import Optional

import codeinfra

class Random(codeinfra.CustomResource):
    def __init__(self,
                 resource_name: str,
                 length: codeinfra.Input[int],
                 prefix: Optional[codeinfra.Input[str]] = None,
                 opts: Optional[codeinfra.ResourceOptions] = None):
        props = {
            "length": length,
            "result": None,
            "prefix": prefix,
        }
        super().__init__("testprovider:index:Random", resource_name, props, opts)

    @property
    @codeinfra.getter
    def length(self) -> codeinfra.Output[int]:
        return codeinfra.get(self, "length")

    @property
    @codeinfra.getter
    def result(self) -> codeinfra.Output[str]:
        return codeinfra.get(self, "result")

    def invoke(self, args):
        return codeinfra.runtime.invoke("testprovider:index:returnArgs", args)


class Component(codeinfra.ComponentResource):
    def __init__(self,
                 resource_name: str,
                 length: codeinfra.Input[int],
                 opts: Optional[codeinfra.ResourceOptions] = None):
        props = {
            "length": length,
            "childId": None,
        }
        super().__init__("testprovider:index:Component", resource_name, props, opts, True)

    @property
    @codeinfra.getter
    def length(self) -> codeinfra.Output[int]:
        return codeinfra.get(self, "length")

    @property
    @codeinfra.getter
    def child_id(self) -> codeinfra.Output[str]:
        return codeinfra.get(self, "childId")

class Provider(codeinfra.ProviderResource):
    def __init__(__self__,
                 resource_name: str,
                 opts: Optional[codeinfra.ResourceOptions] = None):
        super(Provider, __self__).__init__(
            'testprovider',
            resource_name,
            None,
            opts)

File: tests/integration/transforms/nodejs/simple/random.ts
// Copyright 2016-2021, KhulnaSoft Ltd.  All rights reserved.

import * as codeinfra from "@codeinfra/codeinfra";

interface RandomArgs {
    length: codeinfra.Input<number>;
    prefix?: codeinfra.Input<string | undefined>;
}

export class Random extends codeinfra.CustomResource {
    public readonly length!: codeinfra.Output<number>;
    public readonly result!: codeinfra.Output<string>;
    constructor(name: string, args: RandomArgs, opts?: codeinfra.CustomResourceOptions) {
        super("testprovider:index:Random", name, args, opts);
    }

    randomInvoke(args) {
	return codeinfra.runtime.invoke("testprovider:index:returnArgs", args);
    }
}


interface ComponentArgs {
    length: codeinfra.Input<number>;
}

export class Component extends codeinfra.ComponentResource {
    public readonly length!: codeinfra.Output<number>;
    public readonly childId!: codeinfra.Output<string>;
    constructor(name: string, args: ComponentArgs, opts?: codeinfra.ComponentResourceOptions) {
        super("testprovider:index:Component", name, args, opts, true);
    }
}

export class TestProvider extends codeinfra.ProviderResource {
    constructor(name: string, opts?: codeinfra.ResourceOptions) {
        super("testprovider", name, {}, opts);
    }
}
